src/js/utils/Util.js

```diff
@@ -208,6 +208,42 @@ const Util = {
     }
 
     return item;
+  },
+
+  isString(str) {
+    return typeof str === "string";
+  },
+
+  /**
+   * Parse url string
+   * and returns its object representation
+   *
+   * @param {String} url
+   * @returns {Object} url representation
+   */
+  parseUrl(url) {
+    if (!Util.isString(url)) {
+      return null;
+    }
+
+    // TODO: replace below with browser native API <new URL()>
+    // when all browsers support are available
+    const aElement = document.createElement("a");
+    aElement.href = url;
+
```

Comment:
the reason I'm returning the object here in this schema is because when moving to use the browser native API it's going to return a obj that way we don't break the API for those using `parseUrl` down the road